node/src/components/small_network.rs

```diff
@@ -445,6 +445,25 @@ where
                 peer_consensus_public_key,
                 stream,
             } => {
+                if self.cfg.max_incoming_peer_connections != 0 {
+                    if let Some(symmetries) = self.connection_symmetries.get(&peer_id) {
+                        let incoming_count = symmetries
+                            .incoming_addrs()
+                            .map(|addrs| addrs.len())
```

Comment:
Just wondering, why would a single peer (IP) try to connect to us on multiple sockets?

Reply:
Bugs, attacks or race conditions due to shutdown/reconnection. All edge cases, basically, that do not occur on well-behaved nodes. Still, this needed an upper bound.